snuba_sdk/function.py

```diff
@@ -103,3 +105,34 @@ class Function(CurriedFunction):
     initializers: Optional[Sequence[Union[ScalarLiteralType, Column]]] = field(
         init=False, default=None
     )
+
+
+identifier_re = re.compile("^[a-zA-Z_][a-zA-Z0-9_]*$")
+
+
+class InvalidLambdaError(InvalidExpressionError):
+    pass
+
+
+@dataclass(frozen=True)
+class Identifier(Expression):
+    name: str
+
+    def validate(self) -> None:
+        if not isinstance(self.name, str) or not identifier_re.match(self.name):
+            raise InvalidLambdaError(f"`{self.name}` is not a valid identifier")
+
+
+@dataclass(frozen=True)
+class Lambda(Expression):
+    identifiers: Sequence[str]
+    transformation: CurriedFunction
+
+    def validate(self) -> None:
+        if not isinstance(self.identifiers, (tuple, list)):
+            raise InvalidLambdaError("identifiers must be a sequence")
+        for i in self.identifiers:
```

Comment:
If self.identifiers is going to be `Identifier` then you can just call `self.identifiers.validate` here.

Reply:
I would like to leave them as strings (see above) but also I couldn't call `self.identifiers.validate` directly since that property is a Sequence which doesn't have a validate function.